### DISCUSSION TITLE
Issues when running Scikit Learn Wheels

### DISCUSSION THREAD

**Discussion by ZAIpty** [JonathanFly/bark#82] [Q&A]
Hi, I am quite new to machine learning and I dont know if something is wrong with my system or is this a bug that is causing the trouble.

![image](https://github.com/JonathanFly/bark/assets/134893653/6fc2d42d-d3f5-4002-87de-22a0baa22cd3)

![image](https://github.com/JonathanFly/bark/assets/134893653/3b390832-7e5a-4273-a245-a0e932763b96)

I have updated C++ in order to run properly Scikit, updated python from 3.10 to 3.11 and also tried to upgrade Setuptools to a 59.8.0 but nothing seems to work. 

Seems to me that the conda environment is setted up on Python 3,10 but Setuptools is on version 67 due to the Numpy clarification below, I guess this may be causing the issue.

_For Python versions older than 3.12, change the version of setuptools you are using in order to work with a specific package, you can do so by specifying the version number when installing or upgrading setuptools. For example, if you need to use setuptools version 59.0, you can run the command pip install --upgrade setuptools==59.0. This will install version 59.0 of setuptools and upgrade any existing version you have installed. Keep in mind that using an older version of setuptools may cause compatibility issues with other packages that require a newer version. It's generally recommended to use the latest version of setuptools unless you have a specific reason to use an older version._

Any help to a new coder in town?

**Comment by JonathanFly** [comment]
Do you need Python 3.11 specifically? 3.11 should work fine if you're in a conda env, it should keep everything separate. But obviously some wires are getting crossed.

But in general, Python 3.10 is the the one that is compatible with everything, including typically faster performance because of better CUDA support.



I think you can start the conda with the `cmd_windows.bat` file. That will start it up alone.

You actually might just an update. Try this first:

```
pip install --upgrade setuptools 
conda update -y conda
conda update --all
conda install setup
```



If that doesn't work you can try uninstalling.
```
pip uninstall setuptools
conda uninstall setuptools
```
etc

**Comment by JonathanFly** [comment]
Not your fault, I just tried this fresh, and get same error. Sigh. Incoming fix.

**Comment by JonathanFly** [comment]
> Hi, I am quite new to machine learning and I dont know if something is wrong with my system or is this a bug that is causing the trouble.
> 
> ![image](https://user-images.githubusercontent.com/134893653/245999944-6fc2d42d-d3f5-4002-87de-22a0baa22cd3.png)
> 
> ![image](https://user-images.githubusercontent.com/134893653/246000135-3b390832-7e5a-4273-a245-a0e932763b96.png)
> 
> I have updated C++ in order to run properly Scikit, updated python from 3.10 to 3.11 and also tried to upgrade Setuptools to a 59.8.0 but nothing seems to work.
> 
> Seems to me that the conda environment is setted up on Python 3,10 but Setuptools is on version 67 due to the Numpy clarification below, I guess this may be causing the issue.
> 
> _For Python versions older than 3.12, change the version of setuptools you are using in order to work with a specific package, you can do so by specifying the version number when installing or upgrading setuptools. For example, if you need to use setuptools version 59.0, you can run the command pip install --upgrade setuptools==59.0. This will install version 59.0 of setuptools and upgrade any existing version you have installed. Keep in mind that using an older version of setuptools may cause compatibility issues with other packages that require a newer version. It's generally recommended to use the latest version of setuptools unless you have a specific reason to use an older version._
> 
> Any help to a new coder in town?

I fixed this in the one click installer.

Not the most elegant fix, I pinned:

```audiolm_pytorch==1.1.4```

To a version right before audiolm_pytorch pinned a specific scikit version. They pinned it because of training reasons, so it should be fine. There's probably a cleaner fix but I just wanted to get this working.

**Reply by ZAIpty** [reply]
This worked for me! I got an error with pip uninstall  markdown-it-py but looks like it left some files behind, and these residual files were causing issues, I removed them and BOOM it worked. Now I am playing with it but looks like the Voice Cloner Hubert quantizer runs solely on CPU? I am having like 12.00s/it and it is killing me, wondering if it could run on GPU, went on [gitmylo’s Hubert Model](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer) but doesnt mention GPU anywere so I wonder if this runs solely on CPU? Maybe?

**Comment by JonathanFly** [comment]
> This worked for me! I got an error with pip uninstall markdown-it-py but looks like it left some files behind, and these residual files were causing issues, I removed them and BOOM it worked. Now I am playing with it but looks like the Voice Cloner Hubert quantizer runs solely on CPU? I am having like 12.00s/it and it is killing me, wondering if it could run on GPU, went on [gitmylo’s Hubert Model](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer) but doesnt mention GPU anywere so I wonder if this runs solely on CPU? Maybe?

The cloning model does run on CPU only, by default, but it's usually super fast so this on purpose. It lets you use much larger audio samples without crashing from running out of GPU memory.

I have a guess though. What might be happening is the cloning is fast, but the generating of samples isn't, and is stuck on CPU mode or something?

Try just generating a regular Sample. If it takes forever you probably have to reinstall, not sure what exactly went wrong.
